File: medplum-java/medplum-java-model/src/main/java/com/fhir/ExplanationOfBenefit_AddItem.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableExplanationOfBenefit_AddItem.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface ExplanationOfBenefit_AddItem {
  Optional<Address> locationAddress();

  Optional<List<ExplanationOfBenefit_Detail1>> detail();

  Optional<List<Reference>> provider();

  Optional<Money> net();

  CodeableConcept productOrService();

  Optional<String> servicedDate();

  Optional<List<positiveInt>> noteNumber();

  Optional<List<ExplanationOfBenefit_Adjudication>> adjudication();

  Optional<List<Extension>> modifierExtension();

  Optional<List<positiveInt>> itemSequence();

  Optional<Period> servicedPeriod();

  Optional<List<CodeableConcept>> subSite();

  Optional<Reference> locationReference();

  Optional<decimal> factor();

  Optional<List<positiveInt>> detailSequence();

  Optional<List<CodeableConcept>> modifier();

  Optional<List<Extension>> extension();

  Optional<Money> unitPrice();

  Optional<List<CodeableConcept>> programCode();

  Optional<CodeableConcept> locationCodeableConcept();

  Optional<Quantity> quantity();

  Optional<CodeableConcept> bodySite();

  Optional<String> id();

  Optional<List<positiveInt>> subDetailSequence();

  static ImmutableExplanationOfBenefit_AddItem.ProductOrServiceBuildStage builder() {
    return ImmutableExplanationOfBenefit_AddItem.builder();
  }
}

File: medplum-typescript/src/model/fhir/ExplanationOfBenefit_AddItem.ts
import { Address } from "./Address";
import { CodeableConcept } from "./CodeableConcept";
import { decimal } from "./decimal";
import { ExplanationOfBenefit_Adjudication } from "./ExplanationOfBenefit_Adjudication";
import { ExplanationOfBenefit_Detail1 } from "./ExplanationOfBenefit_Detail1";
import { Extension } from "./Extension";
import { Money } from "./Money";
import { Period } from "./Period";
import { positiveInt } from "./positiveInt";
import { Quantity } from "./Quantity";
import { Reference } from "./Reference";

export interface ExplanationOfBenefit_AddItem {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    itemSequence: positiveInt[] | null | undefined;
    detailSequence: positiveInt[] | null | undefined;
    subDetailSequence: positiveInt[] | null | undefined;
    provider: Reference[] | null | undefined;
    productOrService: CodeableConcept;
    modifier: CodeableConcept[] | null | undefined;
    programCode: CodeableConcept[] | null | undefined;
    servicedDate: string | null | undefined;
    servicedPeriod: Period | null | undefined;
    locationCodeableConcept: CodeableConcept | null | undefined;
    locationAddress: Address | null | undefined;
    locationReference: Reference | null | undefined;
    quantity: Quantity | null | undefined;
    unitPrice: Money | null | undefined;
    factor: decimal | null | undefined;
    net: Money | null | undefined;
    bodySite: CodeableConcept | null | undefined;
    subSite: CodeableConcept[] | null | undefined;
    noteNumber: positiveInt[] | null | undefined;
    adjudication: ExplanationOfBenefit_Adjudication[] | null | undefined;
    detail: ExplanationOfBenefit_Detail1[] | null | undefined;
}
